### PR TITLE
[partial update] Add some state in RowsetUpdateState and RowsetTxnMetaPB

### LINKED ISSUE #2465: [partial update] Add some state in RowsetUpdateState and RowsetTxnMetaPB
Some first step code to support moving heavy logic out of the apply phrase

### PR DESCRIPTION
**Pull request by decster**

### CODE REVIEW

**Comment by decster** [comment]
run starrocks_be_unittest

**Review by sduzh — DISMISSED** [review]

File: be/src/storage/rowset_update_state.h [hunk 1]
```diff
@@ -46,6 +47,16 @@ class RowsetUpdateState {
     size_t _memory_usage = 0;
     int64_t _tablet_id = 0;
 
+    // states for partial update
+    EditVersion _read_version;
+    uint32_t _next_rowset_id = 0;
+    struct PartialUpdateState {
+        vector<uint64_t> src_rss_rowids;
```
Comment (chaoyli) [inline be/src/storage/rowset_update_state.h:54]: What's the difference between rssids and rss_rowids?

File: gensrc/proto/olap_file.proto [hunk 2]
```diff
@@ -66,6 +66,13 @@ enum SegmentsOverlapPB {
 message RowsetTxnMetaPB {
     repeated uint32 partial_update_column_ids = 1;
     repeated uint32 partial_update_column_unique_ids = 2;
+
+    message FooterInfo {
```
Comment (chaoyli) [inline gensrc/proto/olap_file.proto:70]: It seems to be redundant with PagePointerPB.

**Review by chaoyli — CHANGES_REQUESTED** [review]

File: be/src/storage/rowset_update_state.h (hunk 1 above)
Comment (decster) [reply be/src/storage/rowset_update_state.h:54]: rssid: rowset segment id  uint32
rss_rowid: <rssid uint32, rowid uint32> uint64

File: gensrc/proto/olap_file.proto (hunk 2 above)
Comment (decster) [reply gensrc/proto/olap_file.proto:70]: Footer is not a page or it is?

Comment (chaoyli) [reply gensrc/proto/olap_file.proto:70]: FooterPointerPB

File: be/src/storage/rowset_update_state.cpp [hunk 3]
```diff
@@ -23,18 +22,16 @@ RowsetUpdateState::RowsetUpdateState() = default;
 
 RowsetUpdateState::~RowsetUpdateState() = default;
 
-Status RowsetUpdateState::load(int64_t tablet_id, Rowset* rowset) {
+Status RowsetUpdateState::load(Tablet* tablet, Rowset* rowset) {
```
Comment (Linkerist) [inline be/src/storage/rowset_update_state.cpp:25]: can be const reference once it is immutable semantics.

**Review by chaoyli — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Pull request merged**